### PR TITLE
Gluster Extension

### PR DESCRIPTION
**Pull request by madanadit**
- Ignore changes from the first commit (simple copy from core Alluxio)

### CODE REVIEW

File: underfs/glusterfs/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# (the “License”). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+alluxio.underfs.glusterfs.GlusterFSUnderFileSystemFactory
```
Comment (apc999) [inline underfs/glusterfs/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory]: new line

File: underfs/glusterfs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-core-common</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <!-- Core Alluxio test dependencies -->
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-core-common</artifactId>
+      <type>test-jar</type>
```
Comment (apc999) [inline underfs/glusterfs/pom.xml]: `<scope>provided</scope>`

Comment (madanadit) [reply underfs/glusterfs/pom.xml]: done

File: underfs/glusterfs/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory (hunk 1 above)
Comment (madanadit) [reply underfs/glusterfs/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory]: done

**Comment by madanadit** [comment]
@apc999

**Pull request merged**